src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfig/EditorConfigStorageLocation`1.cs

```diff
@@ -28,6 +29,15 @@ internal sealed class EditorConfigStorageLocation<T> : OptionStorageLocation2, I
         private readonly Func<string, Type, Optional<T>> _parseValue;
         private readonly Func<T, OptionSet, string?> _getEditorConfigStringForValue;
 
+        public EditorConfigStorageLocation(EditorConfigData<T> editorConfigData)
```

Comment:
discussed offline - 2 changes for a later PR
1.  Replace the function variables with just the editorconfig data object and remove the other ctors.
2.  This will allow us to remove Optional<T> from the editorconfigdata object and change the interface to TryGetValueFromString with a normal T